src/components/AppRepoList/AppRepoButton.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from "react";
+import * as Modal from "react-modal";
+import { Redirect } from "react-router";
+
+interface IAppRepoFormProps {
+  name: string;
+  url: string;
+  message?: string;
+  redirectTo?: string;
+  install: (name: string, url: string) => Promise<any>;
+  update: (values: { name?: string; url?: string }) => void;
+  onAfterInstall?: () => Promise<any>;
+}
+
+export const AppRepoForm = (props: IAppRepoFormProps) => {
+  const { name, url, update, install, onAfterInstall } = props;
+  const handleInstallClick = async () => {
+    await install(name, url);
+    if (onAfterInstall) {
+      await onAfterInstall();
+    }
+  };
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    update({ name: e.target.value });
+  const handleURLChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    update({ url: e.target.value });
+  return (
+    <div className="app-repo-form">
+      <h1>Add an App Repository</h1>
+      <label>
+        Name:
+        <input type="text" value={name} onChange={handleNameChange} />
+      </label>
+      <label>
+        URL:
+        <input type="text" value={url} onChange={handleURLChange} />
+      </label>
+      <button className="button button-primary" onClick={handleInstallClick}>
+        Install Repo
+      </button>
+      {props.redirectTo && <Redirect to={props.redirectTo} />}
+    </div>
+  );
+};
+
+interface IAppRepoAddButtonProps {
+  install: (name: string, url: string) => Promise<any>;
+  redirectTo?: string;
+}
+interface IAppRepoAddButtonState {
+  error?: string;
+  modalIsOpen: boolean;
+  name: string;
+  url: string;
+}
+
+export class AppRepoAddButton extends React.Component<
```

Comment:
Could you split this file into two ones? Just to have 1 component per file. If this component is only used by the above one, you can scope them inside a folder and move the main one to `folder/index.tsx`.